### ISSUE TITLE
2.113.34 CDS Parser unknown type issue. Caused by #3446

### ISSUE THREAD

**Issue by DGRHicron** [abaplint/abaplint#3448]
Hi,
the newest version 2.113.34 causes some CDS related issues. The CDS Parser doesn't report any error in my CDSes but in objects which use CDS as type I get the following issue:
`Contains unknown, CDS parser error, unknown source (unknown_types) [E]`

Quick example what I mean:
```
DATA foo TYPE i_salesdocument.
WRITE foo-salesdocument.
```

**Comment by larshp** [comment]
thanks, I'll look into it tomorrow

**Comment by larshp** [comment]
@DGRHicron can you share some sample code?

**Comment by DGRHicron** [comment]
> @DGRHicron can you share some sample code?

Unfortunately, nothing more than I already shared.
Have you tried already to declare a CDS and use it in a class's method as type of a parameter?

**Comment by larshp** [comment]
just need a bit about the structure and stuff used in the DDLS, try reproducing with dummy fields/tables?

**Comment by DGRHicron** [comment]
@larshp 
I hope this will help: https://github.com/DGRHicron/abaplint_bugs

**Comment by larshp** [comment]
try again, 2.113.37 published

**Comment by DGRHicron** [comment]
@larshp Hmmm... it works on my example repo (Z namespace) but original repo still has the same issues even though I can see that abaplint version used is right (2.113.37). Could it be the namespace related? Original repo uses /xxx/ pattern of the namespace.

**Comment by DGRHicron** [comment]
OK. That's probably not a namespace issue as this exact example (fi_position related) is ok on the main repo. Just rest of the issues (other CDSes related) didn't vanish as I would expect.

**Comment by DGRHicron** [comment]
OK! I see another issue. CDS Parser doesn't understand when a field is from consumed annotation. Like this:
```
_BillingDocument.BillingDocument
```

**Comment by DGRHicron** [comment]
@larshp  I pushed the example to https://github.com/DGRHicron/abaplint_bugs

**Comment by larshp** [comment]
> @larshp I pushed the example to https://github.com/DGRHicron/abaplint_bugs

https://github.com/DGRHicron/abaplint_bugs/blob/main/src/zi_header.ddls.asddls#L20 is this line valid? "_JournalEntryItem" doesnt seem like its configured anywhere?

**Comment by larshp** [comment]
@DGRHicron try again, 2.113.40 published

**Comment by DGRHicron** [comment]
When I created this issue I had 75 issues related to the CDS parser. After 2.113.37 I had 63. After 2.113.40 I have 52. I will now put all objects related to issues to my example repo at once. Shall I continue the topic here?

**Comment by larshp** [comment]
> When I created this issue I had 75 issues related to the CDS parser. After 2.113.37 I had 63. After 2.113.40 I have 52. I will now put all objects related to issues to my example repo at once. Shall I continue the topic here?

yea, feel free, step by step, day by day

**Comment by larshp** [comment]
some CDS fixes in https://github.com/abaplint/abaplint/pull/3464

plus I added CDS to the playground, https://playground.abaplint.org

**Comment by DGRHicron** [comment]
> plus I added CDS to the playground, https://playground.abaplint.org

That is wonderful news! I missed it a lot.

**Comment by DGRHicron** [comment]
Hah, still 52 issues. Nothing has changed. I am working on the example repository. I will tell you once it's done.

**Comment by DGRHicron** [comment]
@larshp for your information - I am returning to this topic. I will probably do the example this week or next week.

**Comment by DGRHicron** [comment]
Done. New examples at [Abaplint Bugs](https://github.com/DGRHicron/abaplint_bugs)

**Comment by DGRHicron** [comment]
Just for the record, Issues still persist on Abaplint 2.113.90

**Comment by larshp** [comment]
@DGRHicron try again, 2.113.91 published

**Comment by DGRHicron** [comment]
> @DGRHicron try again, 2.113.91 published

All issues fixed! Thanks!

**Comment by DGRHicron** [comment]
@larshp, on version 2.113.97 and 2.113.98 I get CDS Parser error again at the following line in zi_taxitem.

`cast( cast( cast( TaxRate as abap.dec( 11, 2 ) ) / 10  as abap.dec(11,2) ) as abap.curr( 23, 2 ) ) as TaxRate,`

Example published on [abaplint_bugs](https://github.com/DGRHicron/abaplint_bugs).

**Comment by larshp** [comment]
@DGRHicron try again, latest version

**Comment by DGRHicron** [comment]
@larshp I confirm. Now there is no issue. Thanks!
